examples/qa_with_sources/server/api.py

```diff
@@ -20,55 +20,29 @@ def __init__(self, **kwargs):
         # set up LLM
         self.llm = OpenAI(client=self.client, temperature=0, cache=True, max_words=250)
         # create a persistent embedding store
-        self.index = self.client.use_plugin(
-            "embedding-index",
-            config={
-                "embedder": {
-                    "plugin_handle": "openai-embedder",
-                    "fetch_if_exists": True,
-                    "config": {
-                        "model": "text-similarity-curie-001",
-                        "dimensionality": 4096,
-                    },
-                }
-            },
-            fetch_if_exists=True,
-        )
+        self.index = SteamshipVectorStore(client=self.client, embedding="text-embedding-ada-002")
 
     @post("index_file")
     def index_file(self, file_handle: str) -> bool:
         text_splitter = CharacterTextSplitter(chunk_size=250, chunk_overlap=0)
-        texts = []
         file = File.get(self.client, handle=file_handle)
-        for block in file.blocks:
-            texts.extend(text_splitter.split_text(block.text))
-
-        # give an approximate source location based on chunk size
-        items = [
-            Tag(client=self.client, text=t, value={"source": f"{file.handle}-offset-{i * 250}"})
-            for i, t in enumerate(texts)
-        ]
+        texts = [text for block in file.blocks for text in text_splitter.split_text(block.text)]
+        metadatas = [{"source": f"{file.handle}-offset-{i * 250}"} for i, text in enumerate(texts)]
```

Comment:
i think in the future including metadata that includes the **actual** offsets from the original files will be a potential win.